Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -101,42 +109,50 @@ def Set_Marshall(ResourceSettings):
                 proxy_conf_path = PROXY_CONF_PATH_LEGACY
+
+            args = ["python", REGISTRATION_FILE_PATH, "--register", "-w", settings.workspace_id, "-a", agent_id,
+                    "-c", OMS_CERTIFICATE_PATH, "-k", OMS_CERT_KEY_PATH, "-f", WORKING_DIRECTORY_PATH, "-s",
+                    WORKER_STATE_DIR, "-e", settings.azure_dns_agent_svc_zone, "-p", proxy_conf_path, "-g",
+                    KEYRING_PATH]
+
             diy_account_id = get_diy_account_id()
             if diy_account_id:
-                args = ["python", REGISTRATION_FILE_PATH, "--register", "-w", settings.workspace_id, "-a", agent_id,
-                        "-c",
-                        OMS_CERTIFICATE_PATH, "-k", OMS_CERT_KEY_PATH, "-f", WORKING_DIRECTORY_PATH, "-s",
-                        WORKER_STATE_DIR,
-                        "-e", settings.azure_dns_agent_svc_zone, "-p", proxy_conf_path, "-g", KEYRING_PATH, "-y",
-                        diy_account_id]
-            else:
-                args = ["python", REGISTRATION_FILE_PATH, "--register", "-w", settings.workspace_id, "-a", agent_id,
-                        "-c", OMS_CERTIFICATE_PATH, "-k", OMS_CERT_KEY_PATH, "-f", WORKING_DIRECTORY_PATH, "-s",
-                        WORKER_STATE_DIR, "-e", settings.azure_dns_agent_svc_zone, "-p", proxy_conf_path, "-g",
-                        KEYRING_PATH]
+                args.append("-y")
+                args.append(diy_account_id)
+
+            asset_tag, is_azure_vm, vm_id = get_optional_metadata()
+            args.append("-i")
+            args.append(vm_id)
+
+            if is_azure_vm:
+                args.append("-z")
+
             proc = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             stdout, stderr = proc.communicate()
             # log(DEBUG, "Trying to register Linux hybrid worker with args: %s" % str(args))
-            if proc.returncode != 0:
-                log(ERROR, "Linux Hybrid Worker registration failed: Return code %s :" % str(proc.returncode)
+            if proc.returncode == -5:
+                log(ERROR, "Linux Hybrid Worker registration failed: DIY and auto-register account ids do not match")
+                log(INFO, "Worker manager with be started without auto registered worker")
+            elif proc.returncode != 0:
+                raise Exception("Linux Hybrid Worker registration failed: Return code %s :" % str(proc.returncode)
                     + stderr + "\n" + stdout)
-                if proc.returncode == -5:
-                    log(ERROR, "Linux Hybrid Worker registration failed: DIY and auto-register agent ids do not match")
-                    log(INFO, "Worker manager with be started without auto registered worker")
+
             elif not os.path.isfile(AUTO_REGISTERED_WORKER_CONF_PATH):
-                log(ERROR, "Linux Hybrid Worker registration file could not be created")
```

Comment:
This line of code below always logs the exception before acting on it. It was changed from only a log to exception because we want to break the execution on the method
except Exception, e:
        log(ERROR, "Set_Marshall returned [-1] with following error: %s" % str(e))
        return [-1]